src/EnergyPlus/General.cc

```diff
@@ -155,6 +155,168 @@ enum class RptKey
+    // See Press et al., Numerical Recipes in Fortran, Cambridge University Press,
+    // 2nd edition, 1992. Page 347 ff.
+
+    // SUBROUTINE ARGUMENT DEFINITIONS:
+    // = -2: f(x0) and f(x1) have the same sign
+    // = -1: no convergence
+    // >  0: number of iterations performed
+
+    Real64 constexpr SMALL(1.e-10);
+
+    Real64 X0;       // present 1st bound
+    Real64 X1;       // present 2nd bound
+    Real64 XTemp;    // new estimate
+    Real64 Y0;       // f at X0
+    Real64 Y1;       // f at X1
+    Real64 YTemp;    // f at XTemp
+    Real64 DY;       // DY = Y0 - Y1
+    bool Conv;       // flag, true if convergence is achieved
+    bool StopMaxIte; // stop due to exceeding of maximum # of iterations
+    bool Cont;       // flag, if true, continue searching
+    int NIte;        // number of interations
+    int AltIte;      // an accounter used for Alternation choice
+
+    X0 = X_0;
+    X1 = X_1;
+    XTemp = X0;
+    Conv = false;
+    StopMaxIte = false;
+    Cont = true;
+    NIte = 0;
+    AltIte = 0;
+
+    Y0 = f(X0);
+    Y1 = f(X1);
+    // check initial values
+    if (Y0 * Y1 > 0) {
+        Flag = -2;
+        XRes = X0;
+        return;
+    }
+    XRes = XTemp;
+
+    while (Cont) {
+
+        DY = Y0 - Y1;
+        if (std::abs(DY) < SMALL) DY = SMALL;
+        if (std::abs(X1 - X0) < SMALL) {
+            break;
+        }
+        // new estimation
+        switch (state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver) {
+        case HVACSystemRootSolverAlgorithm::RegulaFalsi: {
+            XTemp = (Y0 * X1 - Y1 * X0) / DY;
+            break;
+        }
+        case HVACSystemRootSolverAlgorithm::Bisection: {
+            XTemp = (X1 + X0) / 2.0;
+            break;
+        }
+        case HVACSystemRootSolverAlgorithm::RegulaFalsiThenBisection: {
+            if (NIte > state.dataRootFinder->HVACSystemRootFinding.NumOfIter) {
+                XTemp = (X1 + X0) / 2.0;
+            } else {
+                XTemp = (Y0 * X1 - Y1 * X0) / DY;
+            }
+            break;
+        }
+        case HVACSystemRootSolverAlgorithm::BisectionThenRegulaFalsi: {
+            if (NIte <= state.dataRootFinder->HVACSystemRootFinding.NumOfIter) {
+                XTemp = (X1 + X0) / 2.0;
+            } else {
+                XTemp = (Y0 * X1 - Y1 * X0) / DY;
+            }
+            break;
+        }
+        case HVACSystemRootSolverAlgorithm::Alternation: {
+            if (AltIte > state.dataRootFinder->HVACSystemRootFinding.NumOfIter) {
+                XTemp = (X1 + X0) / 2.0;
+                if (AltIte >= 2 * state.dataRootFinder->HVACSystemRootFinding.NumOfIter) AltIte = 0;
+            } else {
+                XTemp = (Y0 * X1 - Y1 * X0) / DY;
+            }
+            break;
+        }
+        default: {
+            XTemp = (Y0 * X1 - Y1 * X0) / DY;
+        }
+        }
+
+        YTemp = f(XTemp);
+
+        ++NIte;
+        ++AltIte;
+
+        // check convergence
```

Comment:
The logic below is a bit too convoluted and uses too many variables. You don't need a `Conv` variable. If you've converged, just `return` and don't make the user follow the logic until the end of the function.